plugins/optimization-detective/class-od-strict-url-metric.php

```diff
@@ -43,34 +45,36 @@ public static function get_json_schema(): array {
 	 * @since 0.6.0
 	 * @see rest_default_additional_properties_to_false()
 	 *
-	 * @param mixed $schema Schema.
-	 * @return mixed Processed schema.
+	 * @phpstan-param JSONSchema $schema
+	 *
+	 * @param array<string, mixed> $schema Schema.
+	 * @return JSONSchema Processed schema.
 	 */
-	private static function set_additional_properties_to_false( $schema ) {
-		if ( ! isset( $schema['type'] ) ) {
-			return $schema;
-		}
-
+	private static function set_additional_properties_to_false( array $schema ): array {
 		$type = (array) $schema['type'];
 
 		if ( in_array( 'object', $type, true ) ) {
 			if ( isset( $schema['properties'] ) ) {
 				foreach ( $schema['properties'] as $key => $child_schema ) {
-					$schema['properties'][ $key ] = self::set_additional_properties_to_false( $child_schema );
+					if ( isset( $child_schema['type'] ) ) {
+						$schema['properties'][ $key ] = self::set_additional_properties_to_false( $child_schema );
+					}
 				}
 			}
 
 			if ( isset( $schema['patternProperties'] ) ) {
 				foreach ( $schema['patternProperties'] as $key => $child_schema ) {
-					$schema['patternProperties'][ $key ] = self::set_additional_properties_to_false( $child_schema );
+					if ( isset( $child_schema['type'] ) ) {
+						$schema['patternProperties'][ $key ] = self::set_additional_properties_to_false( $child_schema );
```

Comment:
What's the rationale for this change? How was the previous implementation problematic?

Reply:
There was no code coverage for the `return` in:

```php
		if ( ! isset( $schema['type'] ) ) {
			return $schema;
		}
```

In practice, the `$schema` should _always_ have a `type` specified given it is enforced by `\OD_URL_Metric::extend_schema_with_optional_properties()`. At the entrypoint where this method is called in `\OD_Strict_URL_Metric::get_json_schema()` it is absolutely defined, which the new typing makes clear. But there is less certainty about the nested properties, so this just moves the `isset(...['type'])` check down below, which has the same effect but also improves coverage.